### PR TITLE
Support exporting the blob-path without having access to the blobs

### PR DESCRIPTION
**Pull request by pbauer**

### CODE REVIEW

File: src/collective/exportimport/serializer.py [hunk 1]
```diff
@@ -326,9 +326,10 @@ def __call__(self):
             return json_compatible(result)
 
     def get_at_blob_path(obj):
-        full_path = obj.getBlob().committed()
-        if not os.path.exists(full_path):
-            return
```
[the remark this answers is not in the document]
Comment (mauritsvanrees) [reply src/collective/exportimport/serializer.py:331]: My thought was: if no blob exists at a path, then it is useless to return this path: importing it would fail because there is nothing.

When I follow the code of the `committed` method in `ZODB/blob.py`, calling `loadBlob`, I see that you actually get a POSKeyError when no file exists on that path. So in hindsight, my file existence check would either pass successfully, or not get called at all...
In other words: it is safe to remove.

File: src/collective/exportimport/serializer.py [hunk 2]
```diff
@@ -326,9 +326,10 @@ def __call__(self):
...
+        oid = obj._p_oid
```
Comment (mauritsvanrees) [inline src/collective/exportimport/serializer.py:329]: @pbauer Here you are getting the oid of the Archetypes object. In `get_dx_blob_path` you do `obj._blob._p_oid`. So there you get the oid of the blob. Shouldn't you do the same in the Archetypes case?
Note: I have not tried it, and don't have a website handy for quickly checking this.

**Review by mauritsvanrees — COMMENTED** [review]
I mostly wonder what the use case is. Ah, maybe you want to copy only the small Data.fs to your laptop, without the multi Gigabyte blobs, and do the export there? If that works, that could be useful.

I suppose it works, although I have a small doubt about the difference between the Archetypes and Dexterity code. I did not try it yet.

The good thing about his PR, even when the blobstorage is available as usual, is that it should be more performant: I think the previous code actually loads the blob in memory.

Theoretically, the previous code is safer, because by sticking to officially supported ways instead of accessing underscore attributes, it makes sure to only load files that have been committed. But I suppose the only way in which the new code could possibly go wrong, is when in one transaction you add a file **and** you call this export code. That seems highly unlikely. So should be fine.

**Comment by pbauer** [comment]
My use-case is a large university that I migrate with really bad VPN connenction, so getting a local copy of their 100GB blobstorage is not possible unless mailing usb drives. This way I can develop and test the migration locally and then run it on their servers without having to download the blobs.

File: src/collective/exportimport/serializer.py (hunk 2 above)
Comment (pbauer) [reply src/collective/exportimport/serializer.py:329]: Ups, good catch, that's a error that I made during copy and paste. The `oid` needs to be from the blob and the `tid` needs to be from the object. 
I guess I should write a test for that feature...

Comment (mauritsvanrees) [reply src/collective/exportimport/serializer.py]: I tried it out in coredev 5.2 with Archetypes, creating both a dexterity and Archetypes file or image, and then trying out the code in `bin/instance-archetypes debug`. The current code actually seems okay, except for one thing, although I guess it works out in the end. In the Archetypes code we call `getBlobFilename`, which returns a full path. In the DX code we call `getBlobFilePath`, which returns the relative path from the blobstorage root. In both cases we then call `get_relative_blob_path` on the result, which gives us the relative path in both cases.
So looks like the code in this PR works, but could be simplified: call `getBlobFilePath` in both cases, and do not call `get_relative_blob_path`.

Comment (pbauer) [reply src/collective/exportimport/serializer.py]: Darn, I forgot that DemoStorage has no filesystem-blobs. I manually tested that it works though and added tests for the other two options...

Comment (pbauer) [reply src/collective/exportimport/serializer.py]: Thanks for the tip with `getBlobFilePath`!

**Review by mauritsvanrees — APPROVED** [review]
LGTM. Thanks!

**Comment by ale-rt** [comment]
I am also using this approach.
The export time is way faster.
Also I am not copying the data all at once but I mount the original blobstorage folder via sshfs.
